phy/cluster/manual/clustering.py

```diff
@@ -113,56 +165,83 @@ def __init__(self, spike_clusters):
         self._spike_clusters_base = self._spike_clusters.copy()
 
     def reset(self):
+        """Reset the clustering to the original clustering.
+
+        All changes are lost.
+
+        """
+        self._undo_stack.clear()
         self._spike_clusters = self._spike_clusters_base
         self._update_all_spikes_per_cluster()
 
     @property
     def spike_clusters(self):
```

Comment:
That being said, `session.clustering.spikes_per_cluster` contains what you want, and is always synchronized with `spike_clusters`.